imap_processing/mag/l0/decom_mag.py

```diff
@@ -71,58 +71,118 @@ def export_to_xarray(l0_data: list[MagL0]):
     ----------
     l0_data: list[MagL0]
         A list of MagL0 datapoints
+
+    Returns
+    -------
+    norm_data : xr.Dataset
+        xarray dataset for generating burst data CDFs
+    burst_data : xr.Dataset
+        xarray dataset for generating burst data CDFs
     """
     # TODO split by mago and magi using primary sensor
     # TODO split by norm and burst
-    norm_data = defaultdict(list)
-    burst_data = norm_data.copy()
+    norm_data = []
+    burst_data = []
 
     for datapoint in l0_data:
         if datapoint.ccsds_header.PKT_APID == Mode.NORMAL:
-            for key, value in dataclasses.asdict(datapoint).items():
-                if key != "ccsds_header":
-                    norm_data[key].append(value)
+            norm_data.append(datapoint)
         if datapoint.ccsds_header.PKT_APID == Mode.BURST:
-            burst_data["SHCOARSE"].append(datapoint.SHCOARSE)
-            burst_data["raw_vectors"].append(datapoint.VECTORS)
+            burst_data.append(datapoint)
+
+    norm_dataset = None
+    burst_dataset = None
+
+    if len(norm_data) > 0:
+        norm_dataset = generate_dataset(norm_data)
+    if len(burst_data) > 0:
+        burst_dataset = generate_dataset(burst_data)
+
+    return norm_dataset, burst_dataset
```

Comment:
Rather than returning these as a two-tuple, would you possibly want to return a single dictionary similar to what you had before?
`datasets["norm"] = norm_dataset`
That might make it easier to extend later with `datasets["outboard"]["norm"]`

Reply:
Yeah, I like that

Reply:
I actually might come back to this when I add the rest of the L1A processing. I don't usually like returning dictionaries like that if I have a known number of outputs, and I think I will come up with something better when I add in the other kinds of processing...